### PR TITLE
Feature kh2 web 1/setup ts project

### PR DESCRIPTION
**Pull request by dima-diachkov**
1. setup create react app with TS
2. delete manifests, default css, tests
3. add node packages:
  - yarn
  - react-router
  - react-redux
  - axios
  - material UI

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/build
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
```
Comment (kerrimov) [inline .gitignore:21]: npm won’t be used in this repo

File: package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "@mui/material": "^5.8.0",
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^13.0.0",
+    "@testing-library/user-event": "^13.2.1",
+    "@types/jest": "^27.0.1",
+    "@types/node": "^16.7.13",
+    "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
+    "axios": "^0.27.2",
+    "emmet": "^2.3.6",
```
Comment (kerrimov) [inline package.json:17]: Why do you added emmet?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
+    "react-redux": "^8.0.1",
+    "react-router": "^6.3.0",
+    "react-scripts": "5.0.1",
+    "typescript": "^4.4.2",
+    "web-vitals": "^2.1.0",
```
Comment (kerrimov) [inline package.json:24]: Do we need web-vitals in the project?

File: README.md [hunk 4]
```diff
@@ -1 +1,4 @@
-WTW it's movie list
\ No newline at end of file
+WTW it's movie list
+
+Install dependencies
```
Comment (kerrimov) [inline README.md:3]: Add more info about your project and use styles for it

File: src/react-app-env.d.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (kerrimov) [inline src/react-app-env.d.ts:1]: Do we need this file here?

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
```
Comment (kerrimov) [inline tsconfig.json:2]: Is that default setup or yours?

Comment (dima-diachkov) [reply tsconfig.json:2]: It's the default setup.

File: package.json (hunk 2 above)
Comment (dima-diachkov) [reply package.json:17]: I thought we discussed this at the meeting. Perhaps I misunderstood. In general, a useful thing in my opinion.

File: src/react-app-env.d.ts (hunk 5 above)
Comment (dima-diachkov) [reply src/react-app-env.d.ts:1]: As far as I understand this file is supported for importing CSS Modules. 
"This relates to import of files with .module.css,.module.scss, and .module.sass extensions."

File: src/setupTests.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
```
Comment (kerrimov) [inline src/setupTests.ts:4]: please remove unnecessary comments

**Review by kerrimov — COMMENTED** [review]
please squash commits

File: src/setupTests.ts (hunk 7 above)
Comment (dima-diachkov) [reply src/setupTests.ts:4]: unnecessary comments have been removed

**Comment by dima-diachkov** [comment]
> please squash commits
commits have been squashed

**Review by kerrimov — APPROVED** [review]

**Pull request merged**